go/processor/floria/state_contract_test.go

```diff
@@ -0,0 +1,460 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package floria
+
+import (
+	"testing"
+
+	"github.com/Fantom-foundation/Tosca/go/tosca"
+	"go.uber.org/mock/gomock"
+)
+
+var (
+	address0x00 = tosca.Address{0x00}
+	address0x01 = tosca.Address{0x01}
+
+	remainingGas = tosca.Gas(10)
+)
+
+func TestStateContract_StateSetBalance(t *testing.T) {
+	tests := map[string]struct {
+		input  []byte
+		gas    tosca.Gas
+		sender tosca.Address
+		err    error
+	}{
+		"outOfGas": {
+			make([]byte, 64),
+			1,
+			tosca.Address{},
+			ErrOutOfGas,
+		},
+		"invalidInput": {
+			make([]byte, 63),
+			callValueTransferGas + remainingGas,
+			tosca.Address{},
+			ErrExecutionReverted,
+		},
+		"sameAddress": {
+			make([]byte, 64),
+			callValueTransferGas + remainingGas,
+			tosca.Address{},
+			ErrExecutionReverted,
+		},
+		"successful": {
+			make([]byte, 64),
+			callValueTransferGas + remainingGas,
+			address0x01,
+			nil,
+		},
+	}
+
+	for testName, test := range tests {
+		t.Run(testName, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			state := tosca.NewMockWorldState(ctrl)
+
+			wantGas := tosca.Gas(0)
+			if test.err == nil {
+				wantGas = remainingGas
+				state.EXPECT().SetBalance(tosca.Address{}, tosca.Value{})
+			}
+
+			gas, err := executeStateSetBalance(state, test.sender, test.input, test.gas)
+			if err != test.err {
+				t.Errorf("unexpected error, want %v, got %v", test.err, err)
+			}
+			if gas != wantGas {
```

Comment:
If error was expected (or err != nil) do not check the rest

Reply:
If an error is expected, it is important to return 0 gas that's why it is checked in all cases.

Reply:
ok, right. When returning `err != nil` value should be zero initialized.